nats-streaming-server.go

```diff
@@ -127,7 +127,7 @@ func parseFlags() (*stand.Options, *natsd.Options) {
 	flag.StringVar(&stanOpts.ClientCert, "tls_client_cert", "", "Path to a client certificate file")
 	flag.StringVar(&stanOpts.ClientKey, "tls_client_key", "", "Path to a client key file")
 	flag.StringVar(&stanOpts.ClientCA, "tls_client_cacert", "", "Path to a client CA file")
-	flag.BoolVar(&stanOpts.EmbedNATS, "embed", true, "Embed the NATS Server")
+	flag.BoolVar(&stanOpts.EmbedNATS, "embed_nats", stand.DefaultEmbedNATS, "Embed the NATS Server")
```

Comment:
No you're right, with the current approach, you would have to potentially pass 3 flags: `-embed_nats -a localhost -p 5222`, so `-server nats://localhost:5222` would be better.

For user/pass, the thing is that we also support config file (from which we read and build nats options). So I could rely on `-server` url, and if no `@` is specified, check options for presence of user/pwd, etc..